File: iOS_demo/UyaliBeautyFaceSDKDemo-OC/UyaliBeautyFaceSDKDemo-OC/ViewController.m
//
//  ViewController.m
//  UyaliBeautyFaceSDKDemo-OC
//
//  Created by S weet on 2023/2/27.
//

#import "ViewController.h"
#import "BeautyFilterController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = [UIColor whiteColor];
    UIButton *button = [[UIButton alloc]initWithFrame:CGRectMake(kScreenWidth/2-50, kScreenHeight/2-100, 100, 50)];
    [self.view addSubview:button];
    [button setTitle:@"美颜滤镜" forState:UIControlStateNormal];
    [button setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [button addTarget:self action:@selector(buttonAction) forControlEvents:UIControlEventTouchUpInside];
}

- (void)buttonAction {
    BeautyFilterController *controller = [BeautyFilterController new];
    controller.modalPresentationStyle = UIModalPresentationFullScreen;
    [self presentViewController:controller animated:YES completion:nil];
}


@end

File: iOS_demo/UyaliBeautyFaceSDKDemo/UyaliBeautyFaceSDKDemo/ViewController.swift
//
//  ViewController.swift
//  UyaliBeautyFaceSDKDemo
//
//  Created by S weet on 2023/1/30.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        view.backgroundColor = .white
        let button1 = UIButton(frame: CGRect(x: view.bounds.width/2-50, y: view.bounds.height/2-100, width: 100, height: 50))
        view.addSubview(button1)
        button1.setTitle("相机美颜", for: .normal)
        button1.setTitleColor(UIColor(named: "color_text"), for: .normal)
        button1.addTarget(self, action: #selector(buttonAction1), for: .touchUpInside)
        
        let button2 = UIButton(frame: CGRect(x: view.bounds.width/2-50, y: view.bounds.height/2-50, width: 100, height: 50))
        view.addSubview(button2)
        button2.setTitle("图片美颜", for: .normal)
        button2.setTitleColor(UIColor(named: "color_text"), for: .normal)
        button2.addTarget(self, action: #selector(buttonAction2), for: .touchUpInside)
    }

    @objc func buttonAction1() {
        let controller = BeautyFilterController()
        controller.modalPresentationStyle = .fullScreen
        present(controller, animated: true)
    }

    @objc func buttonAction2() {
        let imagePicker = UIImagePickerController()
        imagePicker.delegate = self
        imagePicker.sourceType = .photoLibrary
        imagePicker.modalPresentationStyle = .fullScreen
        present(imagePicker, animated: true)
    }

}



extension ViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
        let image: UIImage!
        image = info[.originalImage]  as? UIImage
        
        picker.dismiss(animated: true) {[self] in
            if image != nil {
                let controller = BeautyImageController()
                controller.image = image
                controller.modalPresentationStyle = .fullScreen
                present(controller, animated: true)
            }
        }
    }
}
